cypress/support/commands.js

```diff
@@ -29,13 +25,69 @@ import { addMatchImageSnapshotCommand } from 'cypress-image-snapshot/command';
 addMatchImageSnapshotCommand();
 
 Cypress.Commands.add('getByDataCy', (selector) => {
-  cy.get(`[data-cy="${selector}"]`);
+  cy.get(`[placeholder="${selector}"]`);
 });
 
-Cypress.Commands.add('register', (email = 'riot@qa.team', username = 'riot', password = '12345Qwert!') => {
-  cy.request('POST', '/users', {
-    email,
-    username,
+Cypress.Commands.add('register', (
+  email = 'riot@qa.team',
+  username = 'riot',
+  password = '12345Qwert!'
+) => {
+  return cy.request('POST', 'https://conduit.mate.academy/api/users', {
+    user: {
+      email,
+      password,
+      username
+    }
+  }).then((response) => ({
+    ...response.body.user,
     password
+  }));
+});
+
+Cypress.Commands.add('login', (user) => {
+  cy.request('POST', 'https://conduit.mate.academy/api/users/login', {
+    user
+  }).then((response) => {
+    expect(response.status).to.eq(200);
+    const user = {
+      username: response.body.user.username,
+      email: response.body.user.email,
+      token: response.body.user.token,
+      bio: response.body.user.bio,
+      image: response.body.user.image,
+      effectiveImage:
+        'https://www.onthisday.com/images/people/homer-simpson.jpg?w=360'
+    };
+    window.localStorage.setItem('user', JSON.stringify(user));
+    cy.setCookie('auth', response.body.user.token);
+  });
+});
+
+Cypress.Commands.add('registerAndLoginUser', () => {
+  cy.registerNewUser().then((user) => {
+    cy.login(user).then(() => user);
+  });
+});
+
+Cypress.Commands.add('createArticle', (title, description, body) => {
+  cy.getCookie('auth').then((token) => {
+    const authToken = token.value;
+
+    cy.request({
+      method: 'POST',
+      url: '/api/articles',
```

Comment:
Ensure that the API endpoint URL for article creation is correct and accessible. If the base URL changes, update this endpoint accordingly.